website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+order to prevent unauthorized cluster access, Nomad requires all certificates
+be signed by the same Certificate Authority (CA). This should be a *private* CA
+and not a public one like [Let's Encrypt][letsencrypt] as any certificate
+signed by this CA will be allowed to communicate with the cluster.
+
+### Certificate Authority
+
+There are a variety of tools for managing your own CA, [like the PKI secret
+backend in Vault][vault-pki], but for the sake of simplicity in this guide
+we'll use [cfssl][]. You can generate a private CA certificate and key with
+[cfssl][]:
+
+```shell
+# Generate the CA's private key and certificate
+cfssl print-defaults csr | cfssl gencert -initca - | cfssljson -bare nomad-ca
+```
+
+The CA key (`nomad-ca-key.pem`) will be used to sign certificates for Nomad
+nodes and must be kept private. The CA certificate (`nomad-ca.pem`) contains
+the public key necessary to validate Nomad certificates and therefore must be
+distributed to every node that requires access.
+
+### Node Certificates
+
+Once you have a CA certifacte and key you can generate and sign the
+certificates Nomad will use directly. TLS certificates commonly use the
+fully-qualified domain name of the system being identified as the certificate's
+Common Name (CN). However, hosts (and therefore hostnames and IPs) are often
+ephemeral in Nomad clusters. They come and go as clusters are scaled up and
+down or outages occur. Not only would signing a new certificate per Nomad node
+be difficult, but using a hostname provides no security or functional benefits
+to Nomad. To fulfill the desired security properties (above) Nomad certificates
+are signed with their region and role such as:
+
+* `client.global.nomad` for a client node in the `global` region
+* `server.us-west.nomad` for a server node in the `us-west` region
+
+To create certificates for the client and server in the cluster from the
+[Getting Started guide][guide-cluster] with [cfssl][] create ([or
+download][cfssl.json]) the following configuration file as `cfssl.json` to
+increase the default certificate expiration time:
+
+```json
+{
+    "signing": {
+        "default": {
+            "expiry": "87600h",
+            "usages": [
+                "signing",
+                "key encipherment",
+                "server auth",
+                "client auth"
+            ]
+        }
+    }
+}
+```
+
+```shell
+# Generate a certificate for the Nomad server
+echo '{}' | cfssl gencert -ca=nomad-ca.pem -ca-key=nomad-ca-key.pem -config=cfssl.json \
+    -hostname="server.global.nomad,localhost" - | cfssljson -bare server
+
+# Generate a certificate for the Nomad client
+echo '{}' | cfssl gencert -ca=nomad-ca.pem -ca-key=nomad-ca-key.pem -config=cfssl.json \
+    -hostname="client.global.nomad,localhost" - | cfssljson -bare client
+
+# Generate a certificate for the CLI
+echo '{}' | cfssl gencert -ca nomad-ca.pem -ca-key nomad-ca-key.pem -profile=client \
+    - | cfssljson -bare cli
+```
```

Comment:
I'd love for this guide to be a self-contained extension of the Getting Started guide. That being said I'm :+1: to more tf examples in our repo and can link to it from somewhere in this doc.